tinydtls package has undeclared dependencies

#### Description

When tinydtls is added to a minimal example, it won't build until some additional modules are enabled manually.

That's not a problem most of the time (when tinydtls is used as a DTLS library, there's bound to be something else that pulls in UDP sockets), but when using it only for its crypto primitives (like in https://github.com/bergzand/libcose/pull/91), that's not the case.

#### Steps to reproduce the issue

In the hello-world example, add `USEPKG += tinydtls` under `QUIET ?= 1` and `make all`

#### Expected results

tinydtls pulls in any dependencies it has, or (if it can't make sane default decisions) errs at dependency checking time, telling the user to make a particular choice.

#### Actual results

    sys/include/net/sock/udp.h:452:10: fatal error: sock_types.h: No such file or directory
      452 | #include "sock_types.h"
          |          ^~~~~~~~~~~~~~

As a workaround, you can add `USEMODULE += gnrc_sock`. That shifs the error to

    pkg/tinydtls/contrib/sock_dtls.c: In function ‘sock_dtls_session_create’:
    pkg/tinydtls/contrib/sock_dtls.c:295:50: error: ‘const union <anonymous>’ has no member named ‘ipv6’; did you mean ‘ipv4’?
      295 |     memcpy(&remote->dtls_session.addr, &ep->addr.ipv6, sizeof(ipv6_addr_t));

which can be mitigated by a `USEMODULE += gnrc_ipv6`.

#### Versions

Tested on Linux with native board and current RIOT master (7430ea98ed196210a17c2ad565e703b5bafd488a).

#### Steps forward

If the respective code sections are necessary for tinyDTLS to be buildable at all, please add dependencies from the pkg to the modules.

Otherwise, ifdef fences around them could allow slimmer builds for tinydtls-as-crypto-primitive cases (but I do see that they are corner cases right now, so "depend on the module with a note that the dependency could probably removed by putting more work into it" would work fine for me as well).

--- a/pkg/tinydtls/Makefile.include
+++ b/pkg/tinydtls/Makefile.include
@@ -7,7 +7,6 @@ ifeq ($(TOOLCHAIN), llvm)
 endif
 
 INCLUDES += -I$(RIOTBASE)/pkg/tinydtls/include
-DIRS += $(RIOTBASE)/pkg/tinydtls/contrib
 
 ifneq (,$(filter tinydtls,$(USEMODULE)))
   INCLUDES += -I$(PKG_BUILDDIR)
@@ -55,4 +54,9 @@ endif
 
 ifneq (,$(filter tinydtls_ecc,$(USEMODULE)))
   DIRS += $(PKG_BUILDDIR)/ecc
+endif
+
+# For now contrib only contains sock_dtls adaption
+ifneq (,$(filter tinydtls_sock_dtls,$(USEMODULE)))
+  DIRS += $(RIOTBASE)/pkg/tinydtls/contrib
 endif

